custombackend/src/main/java/com/algolia/custombackend/elasticbackend/ElasticTransformer.java

```diff
@@ -0,0 +1,272 @@
+
+        SearchResultsHandler<JSONObject> completionHandler;
+
+        AsyncRequest(@NonNull SearchResultsHandler<JSONObject> completionHandler) {
+            this.completionHandler = completionHandler;
+        }
+
+        @Override
+        public void cancel() {
+
+        }
+
+        @Override
+        public boolean isFinished() {
+            return false;
+        }
+
+        @Override
+        protected JSONObject doInBackground(JSONObject... elasticSearchParameters) {
+            InputStream stream = null;
+            HttpURLConnection hostConnection = null;
+            JSONObject searchParameters = elasticSearchParameters[0];
+            try {
+                // Build URL.
+                String urlString = "https://tests-first-sandbox-9472672183.eu-west-1.bonsaisearch.net/concerts/_search?";
+                URL hostURL = new URL(urlString);
+
+                String basicAuth = null;
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+                    basicAuth = "Basic " + new String(android.util.Base64.encode("3nmp9kz7fh:gch2ewzerx".getBytes(), android.util.Base64.NO_WRAP));
+                }
+
+                // Open connection.
+                hostConnection = (HttpURLConnection) hostURL.openConnection();
+                hostConnection.setRequestProperty ("Authorization", basicAuth);
+
+                // Headers
+                hostConnection.setRequestProperty("Accept-Encoding", "gzip");
+                hostConnection.setRequestProperty("Content-Type","application/json");
+                hostConnection.setRequestMethod("POST");
+
+                DataOutputStream printout = new DataOutputStream(hostConnection.getOutputStream());
+                printout.writeBytes(searchParameters.toString());
+                printout.flush ();
+                printout.close ();
+
+                // read response
+                int code = hostConnection.getResponseCode();
+                final boolean codeIsError = code / 100 != 2;
+                stream = codeIsError ? hostConnection.getErrorStream() : hostConnection.getInputStream();
+                if (stream == null) {
+                    throw new IOException(String.format("Null stream when reading connection (status %d)", code));
+                }
+
+                final byte[] rawResponse;
+                String encoding = hostConnection.getContentEncoding();
+                if (encoding != null && encoding.equals("gzip")) {
+                    rawResponse = _toByteArray(new GZIPInputStream(stream));
+                } else {
+                    rawResponse = _toByteArray(stream);
+                }
+
+                // handle http errors
+                if (codeIsError) {
+                    if (code / 100 == 4) {
+                        throw new AlgoliaException(_getJSONObject(rawResponse).getString("message"), code);
```

Comment:
nit: would use regular `RuntimeException` (or standard subclass) instead of our `AlgoliaException` , especially as you catch `Exception` anyway ¯\_(ツ)_/¯

Reply:
Done